Add configuration properties to documentation (#1040)
microsoft/vscode-react-native: README.md

See [Setting up debug environment](doc/debugging.md) for more details.

## React Native debug configuration properties
Name |Description|Type|Defaults|
|---|---|---|---|

> |`cwd`|The path to project root folder|`string`|`${workspaceFolder}`|

Reviewer: ```suggestion
|`cwd`|The path to the project root folder|`string`|`${workspaceFolder}`|
```

|`sourceMaps`|Whether to use JavaScript source maps to map the generated bundled code back to its original sources|`boolean`|`true`|

> |`sourceMapPathOverrides`|A set of mappings for rewriting the locations of source files from what the source map says, to their locations on disk. See https://github.com/Microsoft/vscode-react-native/blob/master/doc/debugging.md#debugging-with-typescript-and-haul for details|`object`|n/a|

Reviewer: ```suggestion
|`sourceMapPathOverrides`|A set of mappings for rewriting the locations of source files from what the source map says, to their locations on disk. See [Debugging with TypeScript and Haul](https://github.com/Microsoft/vscode-react-native/blob/master/doc/debugging.md#debugging-with-typescript-and-haul) for details|`object`|n/a|
```

> |`trace`|Setup logging level in debugger|`log`|

Reviewer: ```suggestion
|`trace`|Logging level in debugger process. May be useful for diagnostics. If set to "Trace" all debugger process logs will be available in `Debug Console` output window|`Log`|
```

> |`address`|TCP/IP address of debug server|`string`|`localhost`|

Reviewer: ```suggestion
|`address`|TCP/IP address of debug server/packager|`string`|`localhost`|
```

> |`address`|TCP/IP address of debug server/packager|`string`|`localhost`|

Reviewer: ```suggestion
|`address`|TCP/IP address of packager to attach to for debugging|`string`|`localhost`|
```

> |`address`|TCP/IP address of debug server (packager)|`string`|`localhost`|

Author: ```suggestion
|`address`|TCP/IP address of packager to attach to for debugging|`string`|`localhost`|
```

> |`port`|Debug port to attach to|`string`|`8081`|

Reviewer: ```suggestion
|`port`| Port of packager to attach to for debugging|`string`|`8081`|
```

|`remoteRoot`|The source root of the remote host|`null`|
|`localRoot`|The local source root that corresponds to the 'remoteRoot'|`${workspaceFolder}`|
|`skipFiles`|An array of file or folder names, or glob patterns, to skip when debugging|`[]`|
|`debuggerWorkerUrlPath`|Path to the app debugger worker to override. For example, if debugger tries to attach to http://localhost:8081/debugger-ui/debuggerWorker.js and you get 404 error from packager output then you may want to change debuggerWorkerUrlPath to another value suitable for your packager (\"debugger-ui\" will be replaced with the value you provide)|`debugger-ui/`|

> |`platform`|The platform to target|n/a|

Reviewer: Please, add possible values here

> |`target`|Device target to run on (either device or simulator)|`simulator`|

Reviewer: The description is incorrect. You can set here `device`, `simulator` and, also, android emulator id, iOS simulator name

> |`logCatArguments`|Arguments to be used for LogCat (The LogCat output will appear on an Output Channel). It can either be an array such as: [\":S\", \"ReactNative:V\", \"ReactNativeJS:V\"] or a string such as \":S ReactNative:V ReactNativeJS:V\"|`["*:S", "ReactNative:V", "ReactNativeJS:V"]`|

Reviewer: ```suggestion
|`logCatArguments`|Arguments to be used for LogCat (The LogCat output will appear on an Output Channel). It can either be an array such as: `[\":S\", \"ReactNative:V\", \"ReactNativeJS:V\"]` or a string such as `\":S ReactNative:V ReactNativeJS:V\"|`["*:S", "ReactNative:V", "ReactNativeJS:V"]`|`
```

Reviewer: Please, double check if we need to escape double quotes for property value.

> |`target`|Target to run on. Possible values: `simulator`, `device`, `<Android device/emulator id>`, `<iOS simulator/device name>`|`string`|`simulator`|

Reviewer: ```suggestion
|`target`|Target to run on. Possible values: `simulator`, `device`, `<Android emulator/device id>`, `<iOS simulator/device name>`|`string`|`simulator`|
```

> |`runArguments`|Run arguments to be passed to `react-native run-<platform>` command(Override all other configuration params)|n/a|

Reviewer: ```suggestion
|`runArguments`|Run arguments to be passed to `react-native run-<platform>` command (override all other configuration params)|n/a|
```

> |`env`|Environment variables passed to the program|`{}`|

Reviewer: Please update the description because it's not clear what this option does.

|`envFile`|Absolute path to a file containing environment variable definitions|`${workspaceFolder}/.env`|

> |`variant`|A variant to be passed to react-native run-android, e.g. 'devDebug' to specify `--variant=devDebug`|n/a|

Reviewer: ```suggestion
|`variant`|A variant to be passed to `react-native run-android`, e.g. use `devDebug` to specify `--variant=devDebug`|n/a|
```

> |`scheme`|A scheme name to be passed to react-native run-ios, e.g. 'devDebug' to specify `--scheme=devDebug`|n/a|

Reviewer: ```suggestion
|`scheme`|A scheme name to be passed to `react-native run-ios`, e.g. `devDebug` to specify `--scheme=devDebug`|n/a|
```

> |`productName`|Bundle display name e.g 'AwesomeProject'|n/a|

Reviewer: Isn't it android only property?

Author: Not it's for setting custom iOS bundle name
